src/Microsoft.VisualStudio.AppDesigner/ApplicationDesigner/ApplicationDesignerView.vb

```diff
@@ -739,6 +742,12 @@ Namespace Microsoft.VisualStudio.Editors.ApplicationDesigner
                         .TabTitle = .EditorCaption
                         .TabAutomationName = PROP_PAGE_TAB_PREFIX & PropertyPages(Index).Guid.ToString("N")
 
+                        Dim TelemetryEvent As TelemetryEvent = New TelemetryEvent(TelemetryEventRootPath + "TabInfo")
```

Comment:
@lifengl Thank you for your comment. I have added the project guid to the telemetry. With respect to sending everything as one event, the problem is once you have that it usually makes the query unusable on the backend (because you will need an mv-expand to unpack the data). Given the relatively low number of pages (and events), I would say the extra cost may not be worth the savings here. What are your thoughts?

Reply:
Discuss offline with Lifeng and we landed on the same opinion that given the number of pages is relatively small and the added complexity on the backend, we are going to be ok here 😊